vid/src/avid_m/proofs.rs

```diff
@@ -82,35 +84,39 @@ impl AvidMScheme {
                     break;
                 }
             }
+            if raw_shares.len() >= param.recovery_threshold {
```

Comment:
if we want "Exactly threshold shares" then maybe we just mark `==` here, and also add a `debug_assert!()` (or a `tracing::error!()`) at the end of the loop for sanity check?

Reply:
there's another check just at the end of the loop